lib/AppInfo/Application.php

```diff
@@ -28,19 +28,19 @@
 use OCP\AppFramework\Bootstrap\IBootContext;
 use OCP\AppFramework\Bootstrap\IBootstrap;
 use OCP\AppFramework\Bootstrap\IRegistrationContext;
-use OCP\AppFramework\IAppContainer;
+use Psr\Container\ContainerInterface;
 
 class Application extends App implements IBootstrap {
 	public function __construct(array $urlParams = []) {
 		parent::__construct('logreader', $urlParams);
 	}
 
 	public function register(IRegistrationContext $context): void {
+		$context->registerService(Formatter::class, function (ContainerInterface $c) {
+			return new Formatter(\OC::$SERVERROOT);
+		});
 	}
 
 	public function boot(IBootContext $context): void {
-		$context->getAppContainer()->registerService(Formatter::class, function (IAppContainer $c) {
-			return new Formatter(\OC::$SERVERROOT);
-		});
```

Comment:
Can you explain this change? Is that an API change from core?

Reply:
Sure. This function is deprecated since Nextcloud 20 in favor of `IRegestrationContext::registerService`:

https://github.com/nextcloud/server/blob/a4dd35e4429e69ab0b6999aef54c6eb0360f395b/lib/public/IContainer.php#L106